autoload/ale/list.vim

```diff
@@ -14,16 +14,19 @@ endfunction
 function! ale#list#SetLists(buffer, loclist) abort
     if g:ale_set_quickfix
         call setqflist(a:loclist)
+        call setqflist([], 'r', {'title': getbufinfo(a:buffer)[0].name})
     elseif g:ale_set_loclist
         " If windows support is off, bufwinid() may not exist.
         if exists('*bufwinid')
             " Set the results on the window for the buffer.
             call setloclist(bufwinid(str2nr(a:buffer)), a:loclist)
+            call setloclist(bufwinid(str2nr(a:buffer)), [], 'r', {'title': getbufinfo(a:buffer)[0].name})
```

Comment:
Won't this clear the list when the title is set? `'r'` replaces the list. I believe you can use this instead to set the list and title in one go.

```vim
call setloclist(
\   bufwinid(str2nr(a:buffer)), 
\   a:loclist, 
\   ' ', 
\   {'title': getbufinfo(a:buffer)[0].name}
\)
```